pkg/controllers/plugin/metrics.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+)
+
+const (
+	metricResultSuccess = "success"
+	metricResultError   = "error"
+
+	metricReasonTemplateFailed = "template_failed"
+	metricReasonDiffFailed     = "diff_failed"
+	metricReasonUpgradeFailed  = "upgrade_failed"
+)
+
+var (
+	pluginReconcileTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "greenhouse_plugin_reconcile_total",
+		},
+		[]string{"pluginDefinition", "clusterName", "plugin", "organization", "result", "reason"})
+)
+
+func init() {
+	metrics.Registry.MustRegister(pluginReconcileTotal)
+}
+
+func updateMetrics(plugin *greenhousev1alpha1.Plugin) {
+	result := metricResultSuccess
+	reason := ""
+
+	for _, condition := range plugin.Status.Conditions {
+		if condition.IsUnknown() {
+			continue
+		}
+
+		switch condition.Type {
+		case greenhousev1alpha1.HelmReconcileFailedCondition:
+			if condition.IsTrue() {
+				result = metricResultError
+				reason = metricReasonTemplateFailed
+			}
+		case greenhousev1alpha1.HelmDriftDetectedCondition:
+			if condition.IsTrue() {
+				result = metricResultError
+				reason = metricReasonDiffFailed
+			}
+		case greenhousev1alpha1.StatusUpToDateCondition:
+			if condition.IsFalse() {
+				result = metricResultError
+				reason = metricReasonUpgradeFailed
+			}
+		}
+	}
+
+	pluginReconcileTotalLabels := prometheus.Labels{
+		"pluginDefinition": plugin.Spec.PluginDefinition,
+		"clusterName":      plugin.Spec.ClusterName,
+		"plugin":           plugin.Name,
+		"organization":     plugin.Namespace,
+		"result":           result,
+		"reason":           reason,
+	}
+	pluginReconcileTotal.With(pluginReconcileTotalLabels).Inc()
+}
```

Comment:
Currently, it is not possible to know why the `HelmReconcileFailedCondition` is set to true. 
Could you instead increase the metrics whenever the condition is set to True, False in the code.

```suggestion
type( 
        metricResult string
        metricReason string
)
const (
	success metricResult = "success"
        error metricResult = "error"


	templateFailed metricReason = "template_failed"
        diffFailed metricReason = "diff_failed"
	upgradeFailed  metricReason = "upgrade_failed"
        uninstallFailed  metricReason = "uninstall_failed"
)


func updateMetrics(plugin *greenhousev1alpha1.Plugin, result metricResult, reason metricReason) {
	pluginReconcileTotalLabels := prometheus.Labels{
		"pluginDefinition": plugin.Spec.PluginDefinition,
		"clusterName":      plugin.Spec.ClusterName,
		"plugin":           plugin.Name,
		"organization":     plugin.Namespace,
		"result":           result,
		"reason":           reason,
	}
	pluginReconcileTotal.With(pluginReconcileTotalLabels).Inc()
}
```

Reply:
Done